marc_to_solr/lib/solr_deleter.rb

```diff
@@ -25,15 +25,19 @@ def delete(ids, batch_size = 100)
     # request payload is XML (even if the response is in JSON via the wt=json param)
     def request_deletion(uri:, body:, content_type: 'text/xml')
       @logger&.info "Deleting #{body}"
-      Faraday.post(uri) do |req|
+      start_time = Time.now
+      response = Faraday.post(uri) do |req|
         req.headers = { "Content-Type" => content_type }
         req.body = body
-        req.options.open_timeout = 10
-        req.options.read_timeout = 10
-        req.options.write_timeout = 10
+        req.options.open_timeout = 60
+        req.options.read_timeout = 60
+        req.options.write_timeout = 60
       end
-    rescue Net::ReadTimeout => net_read_timeout
-      Rails.logger.warn("Failed to transmit the POST request to #{uri}: #{body}")
+      @logger&.info("Delete completed in #{Time.now - start_time} seconds. URL: #{uri}")
+      response
+    rescue Faraday::TimeoutError
+      @logger&.warn("Delete timed out after #{Time.now - start_time} seconds. URL: #{uri} : #{body}")
```

Comment:
Notice that I am using `@logger` rather than `Rails.logger` since this part of the code does not run as a Rails app.